### PR TITLE
GeometryUtils add vector utils

### PR DESCRIPTION
**Pull request by Gustl22**
+ calc bisections
+ sum, diff
+ distance
+ dot product
+ intersections (line/line and line/plane)
+ normalizations
+ normal of plane
+ scale

Each float array is used as one vector (no buffer arrays)
If I overlooked a util that is already implemented let me know.

### CODE REVIEW

**Comment by devemux86** [comment]
Thanks!
Is it for future S3DB improvements?

**Comment by Gustl22** [comment]
Yes, this refers to S3DB too ;D

File: vtm/src/org/oscim/utils/geom/GeometryUtils.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /*
  * Copyright 2012, 2013 Hannes Janetzek
+ * Copyright 2017 Gustl22
```
Comment (devemux86) [inline vtm/src/org/oscim/utils/geom/GeometryUtils.java]: Probably 2018. :slightly_smiling_face:

**Comment by devemux86** [comment]
Thanks!
Merged via 1d8cdd7febc6d8849f7b1b08cb264a993588e9b0.

**Pull request closed without merging**